### PR TITLE
chore: use ruff for linting and code formatting

### LINKED ISSUE #241: Consider using Ruff to replace flake8, isort, PyLint and Black
[Ruff](https://docs.astral.sh/ruff/) is a "multi-purpose" tool for Python linting and code-formatting which can replace the current tools apart from mypy. This would improve the developer experience as there would be only two tools for linting and formatting instead of four that there is now. Ruff is also very fast which makes it pleasant to use.

There are some downsides too: Ruff does not yet have all the rules which flake8 and PyLint  includes. Also it is still under somewhat active development meaning that previously working code might be "broken" with a newer Ruff version. This problem can of course be avoided by using a pinned version which probably should be done anyway.

Would this idea make sense?

### PR DESCRIPTION
**Pull request by terop**
ruff replaces black, flake8, isort and pylint.

Fixes #241.

### CODE REVIEW

**Comment by terop** [comment]
I have enabled some rules but more can still be enabled if desired. All rules can be found at https://docs.astral.sh/ruff/rules/.

**Comment by terop** [comment]
@ttu could you kindly take a look at this at some point?

File: pyproject.toml [hunk 1]
```diff
@@ -53,35 +50,17 @@ changelog = "https://github.com/ttu/ruuvitag-sensor/blob/master/CHANGELOG.md"
...
-    "bare-except",
-    "broad-except",
-    "broad-exception-raised",
-    "fixme",
-    "dangerous-default-value",
-    "too-few-public-methods",
-    "useless-object-inheritance"
-]
+[tool.ruff.lint]
+select = ["A", "E", "F", "G", "I", "PERF", "PIE", "PL", "PTH", "SIM", "RUF", "Q", "W"]
```
Comment (ttu) [inline pyproject.toml:58]: Minor change could be adding full name of the rule in comment:

```yaml
select = [
    "A",     # flake8-builtins
    "C90",   # maccabe
    "E",     # pycodestyle errors
    "F",     # pyflakes
    "G",     # flake8-logging-format
    "I",     # isort
    "PERF",  # perflint
    "PIE",   # flake8-pie
    "PL",    # pylint
    "PTH",   # flake8-use-pathlib
    "Q",     # flake8-quotes
    "RUF",   # ruff-specific rules
    "SIM",   # flake8-simplify
    "W",     # pycodestyle warnings
]
ignore = [
    "PLR2004", # Magic value used in comparison
    "RUF006",  # Store a reference to the return value of asyncio.create_task
]
```
`"C90",   # maccabe` was also missing from the list

**Review by ttu — CHANGES_REQUESTED** [review]
Thanks, good changes!

I renamed the PR from ci to chore as this affects CI, build, code style etc.

File: pyproject.toml (hunk 1 above)
Comment (terop) [reply pyproject.toml]: Sure, changed.

**Review by ttu — APPROVED** [review]

**Pull request merged**